File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l_p = 0
        r_p = len(nums) - 1
        
        while l_p <= r_p:
            mid = (l_p + r_p) // 2
            if nums[mid] == target:
                return mid
            elif nums[mid] < target:
                l_p = mid + 1
            else:
                r_p = mid - 1
        
        return -1
        

File: Data Structures & Algorithms/rotate-array/submission-0.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int n = nums.size();
        if (n == 0) return;

        k %= n;

        // reverse whole array
        reverse(nums.begin(), nums.end());

        // reverse first k elements
        reverse(nums.begin(), nums.begin() + k);

        // reverse remaining n-k elements
        reverse(nums.begin() + k, nums.end());
    }
};